packages/twenty-server/src/engine/core-modules/twenty-config/dynamic-driver.base.ts

```diff
@@ -0,0 +1,73 @@
+import { createHash } from 'crypto';
+
+import { ConfigVariables } from 'src/engine/core-modules/twenty-config/config-variables';
+import { ConfigVariablesGroup } from 'src/engine/core-modules/twenty-config/enums/config-variables-group.enum';
+import { TwentyConfigService } from 'src/engine/core-modules/twenty-config/twenty-config.service';
+import { TypedReflect } from 'src/utils/typed-reflect';
+
+export abstract class DynamicDriverBase<TDriver> {
+  private currentDriver: TDriver | null = null;
+  private currentConfigKey: string | null = null;
+
+  constructor(protected readonly twentyConfigService: TwentyConfigService) {}
+
+  protected getCurrentDriver(): TDriver {
+    const configKey = this.buildConfigKey();
+
+    if (this.currentConfigKey !== configKey) {
+      if (this.currentDriver) {
+        this.cleanupDriver(this.currentDriver);
+      }
+
+      try {
+        this.currentDriver = this.createDriver();
+      } catch (error) {
+        throw new Error(
+          `Failed to create driver for ${this.constructor.name} with config key: ${configKey}. Original error: ${error instanceof Error ? error.message : String(error)}`,
+        );
+      }
+
+      this.currentConfigKey = configKey;
+    }
+
+    if (!this.currentDriver) {
+      throw new Error(
+        `Failed to create driver for ${this.constructor.name} with config key: ${configKey}`,
+      );
+    }
+
+    return this.currentDriver;
+  }
+
+  protected getConfigGroupHash(group: ConfigVariablesGroup): string {
+    const groupVariables = this.getConfigVariablesByGroup(group);
+
+    const configValues = groupVariables
+      .map((key) => `${key}=${this.twentyConfigService.get(key)}`)
+      .sort()
+      .join('|');
+
+    return createHash('sha256')
+      .update(configValues)
+      .digest('hex')
+      .substring(0, 16);
+  }
+
+  private getConfigVariablesByGroup(
+    group: ConfigVariablesGroup,
+  ): Array<keyof ConfigVariables> {
+    const metadata =
+      TypedReflect.getMetadata('config-variables', ConfigVariables) ?? {};
+
+    return Object.keys(metadata)
+      .filter((key) => metadata[key]?.group === group)
+      .map((key) => key as keyof ConfigVariables);
+  }
+
+  protected abstract buildConfigKey(): string;
+  protected abstract createDriver(): TDriver;
+  protected cleanupDriver(_driver: TDriver): void {
```

Comment:
I would advocate to remove this kind of thing since it's not used anywhere for now and the PR is fairly complex